### PR TITLE
 Bookmark Patch 2.1.1

### PR DESCRIPTION
**Pull request by aryanrsuri**
feat 2.1.1
close #82

### CODE REVIEW

**Comment by aryanrsuri** [comment]
# I dont know why workspace.xml is being changed drastically. I changed editors to Webstorm

**Comment by Miniland1333** [comment]
Is the test page as usual https://chem.libretexts.org/Under_Construction/Sandboxes/Student_Programmers/Aryan_Suri/Progress/Sidebar_REMAKE?

**Comment by Miniland1333** [comment]
I have a mini-review for you when testing on my iPhone:

1. The hypothesis bar is overlapping your header. Options for remediation include making your header like 80% width, hiding hypothesis, or moving to the bottom.
2. It is not obvious how to close the sidebar on mobile.
3. There seems to be some scroll bleed-through where scrolling affects the page behind the sidebar. It appears this is on pages that are shorter than the viewport so do not have scrolling enabled.

-Henry

**Comment by aryanrsuri** [comment]
# After looking at my commits, this is more so a pr to fix the bookmarking to add remove, as only one aspect of mobile has been updated. Mobile is likely a long term over many PR thing. I think it should be good to go

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,190 @@
+{
```
Comment (Miniland1333) [inline package-lock.json:1]: This file is on the root path and is likely redundant.

File: public/Aryan Suri/Sidebar/sidebar.js [hunk 2]
```diff
@@ -770,15 +790,19 @@ function saveBookmark() {
...
         createBookmarks();
     }
-    else {
-        console.log("one bookmark");
-    }
 }
 function createBookmarks() {
     const LI = document.createElement("li");
     const URL = sessionStorage.getItem("Bookmark");
     const TITLE = sessionStorage.getItem("Title");
```
Comment (Miniland1333) [inline public/Aryan Suri/Sidebar/sidebar.js:797]: Just as an FYI, you can store complex data in session/local storage such as objects and arrays. If you ever need to store lots of items these structures can do the job with I believe JSON.stringify and JSON.parse.

File: public/Aryan Suri/Sidebar/sidebar.js [hunk 3]
```diff
@@ -770,15 +790,19 @@ function saveBookmark() {
...
-    let INNER = `<div> <p>Bookmark: <a style="display: unset;" href="${URL}"> ${TITLE}</a></p></div>`;
-    LI.innerHTML = INNER;
-    URL ? document.querySelector("#bm-list")?.appendChild(LI) : console.log("No bookmarks");
+    LI.id = "sbBookmark";
+    LI.innerHTML = `<div > <p><a style="display: unset;" href="${URL}"> ${TITLE}</a><a id="removeBookmark" style="display: unset;" href="#0" onclick="removeBookmarks()">| Remove</a> </p></div>`;
```
Comment (Miniland1333) [inline public/Aryan Suri/Sidebar/sidebar.js:799]: The remove bookmark link is active and does do a soft page redirect (which is unnecessary and undesired). Either remove the href or apply some sort of e.preventDefault().

**Review by Miniland1333 — CHANGES_REQUESTED** [review]
The only things I caught were:

1. extraneous package-lock.json on the root path
2. remove bookmark link is still active

Once these two things are addressed then it should be good.
-Henry

File: public/Aryan Suri/Sidebar/sidebar.js (hunk 2 above)
Comment (aryanrsuri) [reply public/Aryan Suri/Sidebar/sidebar.js:797]: will def use json once we do mulitple bookmarks

**Comment by Miniland1333** [comment]
Do you know why removing the bookmark closes the panel? I thought it was the redirect causing it but that does not appear to be the case.

**Review by Miniland1333 — DISMISSED** [review]

**Review by Miniland1333 — APPROVED** [review]
No obvious residual problems. PR Approved.
-Henry

**Pull request merged**